pkgs/cli/test/integration.zig

```diff
@@ -0,0 +1,171 @@
+            switch (term) {
+                .Exited => |code| std.debug.print("ERROR: Process exited with code {}\n", .{code}),
+                .Signal => |sig| std.debug.print("ERROR: Process killed by signal {}\n", .{sig}),
+                .Stopped => |sig| std.debug.print("ERROR: Process stopped by signal {}\n", .{sig}),
+                .Unknown => |code| std.debug.print("ERROR: Process terminated with unknown code {}\n", .{code}),
+            }
+        } else {
+            std.debug.print("INFO: Process is still running\n", .{});
+        }
+    }
+
+    return server_ready;
+}
+
+test "CLI beam command with mock network - complete integration test" {
+    const allocator = std.testing.allocator;
+
+    // Verify executable exists first
+    const exe_file = std.fs.openFileAbsolute(build_options.cli_exe_path, .{}) catch |err| {
+        std.debug.print("ERROR: Cannot find executable at {s}: {}\n", .{ build_options.cli_exe_path, err });
+
+        // Try to list the directory to see what's actually there
+        std.debug.print("INFO: Attempting to list zig-out/bin directory...\n", .{});
+        const dir_path = std.fs.path.dirname(build_options.cli_exe_path);
+        if (dir_path) |path| {
+            var dir = std.fs.openDirAbsolute(path, .{ .iterate = true }) catch |dir_err| {
+                std.debug.print("ERROR: Cannot open directory {s}: {}\n", .{ path, dir_err });
+                return err;
+            };
+            defer dir.close();
+
+            var iterator = dir.iterate();
+            std.debug.print("INFO: Contents of {s}:\n", .{path});
+            while (try iterator.next()) |entry| {
+                std.debug.print("  - {s} (type: {})\n", .{ entry.name, entry.kind });
+            }
+        }
+
+        return err;
+    };
+    exe_file.close();
+    std.debug.print("INFO: Found executable at {s}\n", .{build_options.cli_exe_path});
+
+    // Start CLI with beam command and mock network - use build option for executable path
+    const args = [_][]const u8{ build_options.cli_exe_path, "beam", "--mockNetwork", "true" };
+    var cli_process = process.Child.init(&args, allocator);
+
+    // Capture stdout and stderr for debugging
+    cli_process.stdout_behavior = .Pipe;
+    cli_process.stderr_behavior = .Pipe;
+
+    defer {
+        _ = cli_process.kill() catch {};
+        _ = cli_process.wait() catch {};
+    }
+
+    // Start the process
+    cli_process.spawn() catch |err| {
+        std.debug.print("ERROR: Failed to spawn process: {}\n", .{err});
+        return err;
+    };
+
+    std.debug.print("INFO: Process spawned successfully with PID\n", .{});
+
+    // Wait for metrics server to be ready using helper function
+    const server_ready = try spinBeamSimNode(&cli_process);
+
+    // Verify server started successfully
+    try std.testing.expect(server_ready);
+
+    // Let the node run for a bit to generate some activity
+    std.time.sleep(2000 * std.time.ns_per_ms);
```

Comment:
abstract this sleep into waitForNodeStart, this can over the time be abstracted to listen for some event that the node can putout on being active, add this comment as well for the fn